tests/loggers/test_base.py

```diff
@@ -212,3 +213,42 @@ def log_metrics(self, metrics, step):
     assert logger.history == {0: {'loss': 0.5623850983416314}}
     logger.close()
     assert logger.history == {0: {'loss': 0.5623850983416314}, 1: {'loss': 0.4778883735637184}}
+
+
+def test_np_sanitization():
+    class CustomParamsLogger(CustomLogger):
+        def __init__(self):
+            super().__init__()
+            self.logged_params = None
+
+        @rank_zero_only
+        def log_hyperparams(self, params):
+            params = self._convert_params(params)
+            params = self._sanitize_params(params)
+            self.logged_params = params
+
+    logger = CustomParamsLogger()
+    np_params = {
+        "np.bool_": np.bool_(1),
+        "np.byte": np.byte(1),
+        "np.intc": np.intc(1),
+        "np.int_": np.int_(1),
+        "np.longlong": np.longlong(1),
+        "np.single": np.single(1),
+        "np.double": np.double(1),
+        "np.csingle": np.csingle(1),
+        "np.cdouble": np.cdouble(1),
+    }
+    sanitized_params = {
+        "np.bool_": True,
+        "np.byte": 1,
+        "np.intc": 1,
+        "np.int_": 1,
+        "np.longlong": 1,
+        "np.single": 1.0,
+        "np.double": 1.0,
+        "np.csingle": "(1+0j)",
+        "np.cdouble": "(1+0j)",
+    }
```

Comment:
```suggestion
    np_params = {
        "np.bool_": np.bool_(1),
        "np.byte": np.byte(2),
        "np.intc": np.intc(3),
        "np.int_": np.int_(4),
        "np.longlong": np.longlong(5),
        "np.single": np.single(6.7),
        "np.double": np.double(8.9),
        "np.csingle": np.csingle(7+2j),
        "np.cdouble": np.cdouble(9+4j),
    }
    sanitized_params = {
        "np.bool_": True,
        "np.byte": 2,
        "np.intc": 3,
        "np.int_": 4,
        "np.longlong": 5,
        "np.single": 6.7,
        "np.double": 8.9,
        "np.csingle": "(7+2j)",
        "np.cdouble": "(9+4j)",
    }
```
great test, can we just in case they are mapped properly to kind of make it as incremental values